src/ClusterBootstrap/services/dashboard/dashboard.yaml

```diff
@@ -1,68 +1,132 @@
-{% if false %}
-apiVersion: v1
-kind: Service
-metadata:
-  name: dashboard
-  namespace: default
-spec:
-  ports:
-  - port: {{cnf["webuiport"]}}
-  selector:
-    app: dashboard
----
-{% endif %}
 kind: DaemonSet
 apiVersion: apps/v1
 metadata:
   name: dashboard
-  namespace: default
 spec:
   selector:
     matchLabels:
-      dashboard-node: pod
+      app: dashboard
   template:
     metadata:
       name: dashboard
       labels:
         app: dashboard
-        dashboard-node: pod
     spec:
-      {% if cnf["dnsPolicy"] %}    
-      dnsPolicy: {{cnf["dnsPolicy"]}}
-      {% endif %}
       nodeSelector:
         dashboard: active
-      {% if true %}  
-      hostNetwork: true
-      {% endif %}
       containers:
       - name: dashboard
-        image: {{cnf["worker-dockerregistry"]}}/{{cnf["dockerprefix"]}}/dashboard:{{cnf["dockertag"]}}
+        image: '{{cnf["worker-dockerregistry"]}}/{{cnf["dockerprefix"]}}/dashboard:{{cnf["dockertag"]}}'
         imagePullPolicy: Always
-        {% if false %}
+        env:
+        - name: TRUST_PROXY
+          value: "true"
         ports:
-          - name: dashboard
-            containerPort: 80
-        {% endif %}
-        imagePullPolicy: Always
+        - name: http
+          containerPort: 80
         volumeMounts:
-        - mountPath: /usr/src/app/config
-          name: dashboardconfig
-        - mountPath: /var/log/dashboard
-          name: dashboardlog
+        - name: config
+          mountPath: /usr/src/app/config
       {% if cnf["private_docker_registry_username"] %}
       imagePullSecrets:
       - name: svccred
       {% endif %}
       volumes:
-      - name: dashboardconfig
-        hostPath:
-          path: /etc/dashboard
-      - name: dashboardlog
-        hostPath:
-          path: /var/log/dashboard
+      - name: config
+        configMap:
+          name: dashboard
       tolerations:
       - key: CriticalAddonsOnly
         operator: Exists
       - key: node-role.kubernetes.io/master
-        effect: NoSchedule            
+        effect: NoSchedule
+
+---
+
+apiVersion: v1
```

Comment:
If you decide to do this, please make it backward compatible. Central dashboard is not yet managed by k8s

Reply:
Yes, I have not touched the non-k8s parts, deploying with pure docker won't be changed.

Reply:
My proposal is like we add a flag to prevent configmap generation for centralized dashboard and write the dashboard config by ourself to configmap.